data/src/main/java/com/example/data/service/WeatherService.kt

```diff
@@ -1,23 +1,19 @@
 package com.example.data.service
 
 import com.example.data.BuildConfig
-import com.example.data.WeatherServiceRequest
-import com.example.data.mappers.transformApi
-import com.example.domain.entities.ResponseWeatherInfo
-import com.example.domain.service.WeatherInterfaceService
+import com.example.data.service.responses.ResponseApi
 import com.example.domain.utils.ServiceResult
 
-class WeatherService : WeatherInterfaceService {
+class WeatherService {
 
     private val generator = WeatherServiceRequest()
 
-    override fun getWeatherCityInfo(): ServiceResult<ResponseWeatherInfo> {
-        val callResponse =
-            generator.createService(WeatherApi::class.java).callWeatherInfo(CITY, BuildConfig.ApiKey, UNITS)
+    fun getWeatherCityInfoToService(): ServiceResult<ResponseApi> {
+        val callResponse = generator.createService(WeatherApi::class.java).callWeatherInfo(CITY, BuildConfig.ApiKey, UNITS)
         try {
             val response = callResponse.execute()
             if (response.isSuccessful) {
-                response.body()?.transformApi()?.let { return ServiceResult.Success(it) }
+                response.body()?.let { return ServiceResult.Success(it) }
             }
             return ServiceResult.Failure(Exception(response.message()))
```

Comment:
you can return if's, there's no need for so many "return"

Reply:
Done